sdk/java/pulumi/src/main/java/com/pulumi/serialization/internal/Converter.java

```diff
@@ -428,7 +428,15 @@ private Object tryConvertObjectInner(
                             argument,
                             TypeShape.extract(extractSetterParameter(setters.get(name)))
                     );
-                    setters.get(name).invoke(builder, convertedArgument);
+                    // Builders generated by Pulumi codegen typically
+                    // null checks with requireNonNull, so invoking
+                    // the builder method with null will throw. But
+                    // Pulumi SDK needs to tolerate missing data, so
+                    // we err on the side of skipping the builder call
+                    // when the argument is null.
+                    if (convertedArgument != null) {
```

Comment:
Note: this might change semantics in a subtle way, because it skips calling setter on @Nullable params as well. Not sure if in practice this will ever make a difference, because I don't think we'll have "dirty" builders, so the field will be null anyway.

I'd guess this will work fine.

Reply:
Yes, I figured that not calling the method leaves the value null as before. So it's the same behavior.